server/api/src/app.py

```diff
@@ -1,27 +1,43 @@
-from flask import Flask, request
-from flask_cors import CORS
+from quart import Quart, request
+from quart_cors import cors
 
+import asyncio
 import classifier
 import os
 
+from delayed_init import delayed_init
+
+
 def create_app():
-    app = Flask(__name__, instance_relative_config=True)
-    CORS(app)
+    app = Quart(__name__)
+    cors(app)
 
     @app.route('/api')
-    def index():
+    async def index():
         return {
             "version": "0.0.1",
             "git.commit": os.getenv("COMMIT", "0000000"),
             "git.branch": os.getenv("BRANCH", ""),
-    }
+            "yagna.available": hasattr(app, "yagna_available") and app.yagna_available,
+            "yagna.ready": hasattr(app, "yagna_ready") and app.yagna_ready,
+        }
 
     @app.route('/api/classify', methods=["POST"])
-    def classify():
-        text = request.form.get("text_file") or request.form.get("text")
+    async def classify():
+        form = await request.form
+        text = form.get("text_file") or form.get("text")
         print(text)
 
         return classifier.classify_text(text)
 
+    @app.before_serving
+    async def init_wait_yagna():
+        if not hasattr(app, "yagna_available"):
```

Comment:
Not need to check here - it's just called at the beginning, before any request  gets handled